include/mapbox/variant.hpp

```diff
@@ -84,25 +84,27 @@ struct MAPBOX_VARIANT_DEPRECATED static_visitor
     ~static_visitor() {}
 };
 
+using type_index_t = unsigned int;
```

Comment:
I think, for up to 256 alternative types variant's discriminatior type may be clearly expressed in terms of `uint_fast8_t` if speed is priority and `uint8_t` otherwise.